function/result_data.go

```diff
@@ -22,7 +21,7 @@ type ResultData struct {
 }
 
 // Equal returns true if the value is equivalent.
-func (d ResultData) Equal(o ResultData) bool {
+func (d *ResultData) Equal(o ResultData) bool {
```

Comment:
Nit: Was there a reason for this change? It now means that this method needs to potentially account for `d == nil`

Reply:
The change was just to make all methods on the `ResultData` struct on a pointer receiver. Have reverted.